specs/MultiProfile.md

```diff
@@ -29,6 +29,21 @@ Users can use this interface to get the cookie manager, which is shared by all W
 with this profile. The cookie manager got from profile (CoreWebView2.Profile.CookieManager) is the
 same as that got from CoreWebView2 (CoreWebView2.CookieManager).
 
+Currently, we already have **CoreWebView2Settings** interface to manage password-autosave and
+general-autofill, but it will work not immediately but after the next navigation, and it can only
+apply for current WebView2, which means if we start a new WebView2 using the same profile, all the
+settings are default values and cannot be set from the profile. By adding password-autosave and
+general-autofill management interfaces in profile, we can manage the properties and they will apply
+immediately if we set a new value, and all WebView2s that created with the same profile can share
+the settings, which means if we change password-autosave or general-autofill property in one WebView2,
+the others with the same profile will also work. And these two property is linked with the same
+properties in ICoreWebView2Settings, so changing one will change the other. Explain it in detail,
+it will work immediately no matter setting the properties in **CoreWebView2Settings** or
+**CoreWebView2Profile**, and when the property is changed in one interface, the same property in
+the other interface is changed as well immediately. So for the WebView2s with the same profile,
+their **IsPasswordAutosaveEnabled** or **IsGeneralAutofillEnabled** property
+in **CoreWebView2Settings** and **CoreWebView2Profile** should always keep in sync.
```

Comment:
The grammar here makes it hard for me to understand what this is trying to say.

I think it's saying that, given
* `webview2A` and `webview2B` are both `CoreWebView2` objects
* `webview2A.Profile == webview2B.Profile`; i.e., they were created from the same profile.

Under these conditions, the following properties are now all synchronized, and changing one will change all the others:
* `webview2A.Settings.IsPasswordAutosaveEnabled`
* `webview2A.Profile.IsPasswordAutosaveEnabled` (which is the same as `webview2B.Profile.IsPasswordAutosaveEnabled`)
* `webview2B.Settings.IsPasswordAutosaveEnabled`

This is true even for webviews created by different processes, as long as they share a profile.

Isn't this a breaking change? Before this API spec, `webview2A.Settings.IsPasswordAutosaveEnabled` and `webview2B.Settings.IsPasswordAutosaveEnabled` were not synchronized. Different `CoreWebView2` objects could have different autosave settings.

Consider an existing app (shipping today) that uses the evergreen WebView and which creates two CoreWebView2 objects that share a profile, and it does

```csharp
// This webview is used for over-the-shoulder authentication scenarios
// and should not autosave passwords because we don't want the child
// account to autosave the parent's passwords.
webview2A.Settings.IsPasswordAutosaveEnabled = false;

// This webview is used for the main app.
webview2B.Settings.IsPasswordAutosaveEnabled = true;

// Double-check that our over-the-shoulder authentication webview is not saving passwords.
Debug.Assert(webview2A.Settings.IsPasswordAutosaveEnabled == false);
```

then the assertion will pass on old versions of WebView2 but will fail on newer versions.

Reply:
Yes, it seems to be a breaking change for these two settings. But not very sure is there any other better solution for this?

Reply:
There's two confusing things:
1. `CoreWebView2Settings.*` should be per WebView2 not per profile. However, these two `CoreWebView2Settings` properties were always per profile. We're sort of "correcting" this now by adding the profile properties.
2. `CoreWebView2Settings.*` are applied on the next navigation not immediately like profile properties. However, these two properties on `CoreWebView2Settings` have already been behaving this way (applying immediately) for over 1.5 years at least since work introducing the `CoreWebView2Profile` related APIs. This was a breaking change that  apparently no one noticed.

The current behavior described in (1) & (2) is good for `CoreWebView2Profile` properties and the breaking change happened so long ago with no complaint that I'd favor leaving all behavior as is, adding the copies of the properties to the `CoreWebView2Profile`, and updating the documentation for the `CoreWebView2Settings` properties to note their strange (for CoreWebView2Settings) behavior and tell folks to use the `CoreWebView2Profile` version of the properties.

Reply:
The confusion is deepened because the first new paragraph is incorrectly describing existing behavior. "Currently, we already have CoreWebView2Settings interface to manage password-autosave and general-autofill, but it will take effect not immediately but after the next navigation, and it can only apply for current WebView2." Neither of these statements is true. Existing behavior is that password properties take effect immediately (not at next navigation) and that the properties take effect for all WebView2s that share the profile (not only for current WebView2).

Fix that paragraph to describe existing behavior properly, and then the Profile properties make much more sense. They are more logical places for existing properties. No functional change.

Reply:
Thanks for your great suggestion! Updated the documentation.